### PR TITLE
Fix that dict-minfreq doesn't work

### PR DESCRIPTION
**Pull request by wolegechu**

### CODE REVIEW

**Comment by wolegechu** [comment]
I am so careless...It's normal now. @alexholdenmiller

File: parlai/core/dict.py [hunk 1]
```diff
@@ -273,14 +274,14 @@ def remove_tail(self, min_freq):
                 # queue up removals since can't mutate dict during iteration
                 to_remove.append(token)
                 # other dicts can be modified as we go
-                idx = self.tok2idx.pop(token)
+                idx = self.tok2inx.pop(token)
```
Comment (alexholdenmiller) [inline parlai/core/dict.py]: is this right?

File: parlai/core/dict.py [hunk 2]
```diff
@@ -273,14 +274,14 @@ def remove_tail(self, min_freq):
...
                 del self.ind2tok[idx]
         for token in to_remove:
             del self.freq[token]
 
     def load(self, filename):
         """Load pre-existing dictionary in 'token[<TAB>count]' format.
-        Initialize counts from other dictionary, or 0 if they aren't included.
+        Initialize counts from other diCtionary, or 0 if they aren't included.
```
Comment (alexholdenmiller) [inline parlai/core/dict.py]: I C what you did here

File: parlai/core/dict.py [hunk 3]
```diff
@@ -306,6 +307,7 @@ def save(self, filename=None, append=False, sort=True):
 
         If ``sort`` (default ``True``), then first sort the dictionary before saving.
         """
+        self.remove_tail(self.min_freq)
```
Comment (alexholdenmiller) [inline parlai/core/dict.py]: maybe it makes sense to put the remove in "sort"?

**Comment by alexholdenmiller** [comment]
Closing this for now--feel free to reopen if you're interested!

**Pull request closed without merging**